### PR TITLE
Fix metrics switch

### PR DESCRIPTION
**Pull request by kavithacm**
### Description

1. Fix metric switch
2. Auto populate measure field in save panel when a saved object is selected (events)
3. Remove commented code
4. Update jest tests

### Issues Resolved
[List any issues this PR will resolve]

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [ ] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: dashboards-observability/public/components/event_analytics/explorer/save_panel/__tests__/__snapshots__/save_panel.test.tsx.snap [hunk 1]
```diff
@@ -785,5 +787,198 @@ exports[`Saved query table component Renders saved query table 1`] = `
...
+                "value": "farenheit (F)",
+              },
+              Object {
+                "text": "meters (m)",
+                "value": "meters (m)",
+              },
+              Object {
+                "text": "kilometers (k)",
+                "value": "kilometers (k)",
+              },
```
Comment (ps48) [inline dashboards-observability/public/components/event_analytics/explorer/save_panel/__tests__/__snapshots__/save_panel.test.tsx.snap:852]: Can you please update this to be generated in runtime. Like done here: https://github.com/opensearch-project/observability/blob/e88ab448e6d3f30032ff5d48189ba0d25bbebf2d/dashboards-observability/public/components/metrics/top_menu/metrics_export_panel.tsx#L135

File: dashboards-observability/public/components/event_analytics/explorer/explorer.tsx [hunk 2]
```diff
@@ -1059,11 +1068,8 @@ export const Explorer = ({
     isLiveTailOnRef.current,
     patternsData,
     viewLogPatterns,
-    isPatternConfigPopoverOpen,
-    patternRegexInput,
```
Comment (ps48) [inline dashboards-observability/public/components/event_analytics/explorer/explorer.tsx:1063]: are these changes required?  this looks like patterns code.

File: dashboards-observability/public/components/event_analytics/explorer/save_panel/save_panel.tsx [hunk 3]
```diff
@@ -52,12 +53,14 @@ export const SavePanel = ({
   curVisId,
   spanValue,
   setSubType,
+  metricMeasure,
   setMetricMeasure,
   setMetricLabel,
+  metricChecked,
 }: ISavedPanelProps) => {
   const [options, setOptions] = useState([]);
   const [checked, setChecked] = useState(false);
```
Comment (ps48) [inline dashboards-observability/public/components/event_analytics/explorer/save_panel/save_panel.tsx:62]: I think this variable is not required. You can just use `metricChecked`

File: dashboards-observability/public/components/event_analytics/explorer/explorer.tsx (hunk 2 above)
Comment (kavithacm) [reply dashboards-observability/public/components/event_analytics/explorer/explorer.tsx:1063]: resolved some merge conflicts n local, might’ve been deleted during that. Thanks for catching it!

File: dashboards-observability/public/components/event_analytics/explorer/save_panel/save_panel.tsx (hunk 3 above)
Comment (kavithacm) [reply dashboards-observability/public/components/event_analytics/explorer/save_panel/save_panel.tsx:62]: checked state value is to handle the switch locally, rather than pulling in the setMetricChecked all the way from explorer.

**Review by ps48 — APPROVED** [review]

File: dashboards-observability/public/components/event_analytics/explorer/explorer.tsx [hunk 4]
```diff
@@ -1059,11 +1068,10 @@ export const Explorer = ({
     isLiveTailOnRef.current,
     patternsData,
     viewLogPatterns,
+    userVizConfigs,
     isPatternConfigPopoverOpen,
     patternRegexInput,
-    userVizConfigs,
```
Comment (ps48) [inline dashboards-observability/public/components/event_analytics/explorer/explorer.tsx:1064]: Can this order be changed? This looks like ordered array.

**Review by rupal-bq — APPROVED** [review]

File: dashboards-observability/public/components/event_analytics/explorer/explorer.tsx (hunk 4 above)
Comment (kavithacm) [reply dashboards-observability/public/components/event_analytics/explorer/explorer.tsx]: fixed it

**Review by sejli — APPROVED** [review]

**Review by joshuali925 — APPROVED** [review]

**Review by mengweieric — APPROVED** [review]
